### DISCUSSION TITLE
i18n

### DISCUSSION THREAD

**Discussion by shaan1974** [FranckFreiburger/vue3-sfc-loader#24] [Ideas]
Hello,

Any plan to support <i18n> any time soon ?

Best regards
Shaan

**Comment by FranckFreiburger** [comment]
Hi,

i18n (https://github.com/intlify/vue-i18n-next ?) should work as-is.
Do you have an example of something that going wrong ?

**Reply by shaan1974** [reply]
Hello,

If i add in my .vue file the tag <i18n> as describe into this page => https://vue-i18n.intlify.dev/guide/advanced/sfc.html

The new labels are not taken in count ( check with "window.i18n.global.messages" )

Maybe i'm doing something wrong.

( is use => https://unpkg.com/vue-i18n@next )

**Reply by FranckFreiburger** [reply]
At the moment, the best I can do it to add a `customBlockHandler` hook that is called for each custom block. Up to you to handle the custom block content.

eg.
```javascript
      customBlockHandler(block, filename, options) {

        if ( block.type !== 'i18n' )
          return;

        return (component) => {

          component.__i18n = JSON.parse(block.content);
        }
      }
```

This will create a new `__i18n` property on the component.

**Reply by FranckFreiburger** [reply]
even better :

```html
<!DOCTYPE html>
<html>
<body>
  <script src="https://unpkg.com/vue@next/dist/vue.runtime.global.prod.js"></script>
  <script src="https://unpkg.com/vue-i18n@next"></script>
  <script src="vue3-sfc-loader.js"></script>
  <script>

    /* <!-- */
    const config = {
      files: {
        '/component.vue': `
          <template>
            {{ $t('hello') }}
          </template>
          <i18n>
          {
            "en": {
             "hello": "hello world!"
            },
            "ja": {
              "hello": "こんにちは、世界！"
            }
          }
          </i18n>
       `
      }
    };
    /* --> */

    const i18n = VueI18n.createI18n();

    const options = {
      moduleCache: { vue: Vue },
      getFile: url => config.files[url],
      addStyle: () => {},
      customBlockHandler(block, filename, options) {

        if ( block.type !== 'i18n' )
          return

        const messages = JSON.parse(block.content);
        for ( let locale in messages )
          i18n.global.mergeLocaleMessage(locale, messages[locale]);
      }
    }

    const app = Vue.createApp(Vue.defineAsyncComponent(() => window['vue3-sfc-loader'].loadModule('/component.vue', options)));

    app.use(i18n);

    app.mount(document.body);

  </script>
</body>
</html>

```

**Reply by shaan1974** [reply]
Many thanks for your anwser.
Do you already propagate this new feature throught JsDelivr or UNPKG ?

**Reply by FranckFreiburger** [reply]
v0.3.1 has been released.

see.
  https://github.com/FranckFreiburger/vue3-sfc-loader/blob/main/docs/examples.md#example-using-sfc-custom-blocks-for-i18n

**Reply by shaan1974** [reply]
Nice job, now i can imagine to create my own tag like "validation" and put all my validations rules related to my validation directives. This could be extended and use in many ways. Or by ex have a tag "form" that handle the creating of a form dynamicly.
